bin/buckinghamshire/add-parish-categories

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env perl
+#
+# This script adds relevant categories for parishes listed in a CSV file by
+# attempting to match them to the existing parish bodies in FMS.
+#
+# The CSV file needs to have 'name' and 'email' columns with the name of
+# the parish council (which should match the existing body) and the email
+# address that reports should be sent to.
+
+use strict;
+use warnings;
+
+BEGIN {    # set all the paths to the perl code
+    use File::Basename qw(dirname);
+    use File::Spec;
+    my $d = dirname(File::Spec->rel2abs($0));
+    require "$d/../../setenv.pl";
+}
+
+use FixMyStreet::DB;
+use Text::CSV;
+
+my $categories = [
+    'Grass cutting',
+    'Hedges',
+    'Fly posting',
+    'Dirty signs',
+];
+
+my @parishes;
+
+my $csv = Text::CSV->new ({ binary => 1, auto_diag => 1 });
+die "Usage: $0 <csv_file>\n" unless @ARGV;
+open my $fh, "<:encoding(utf8)", $ARGV[0] or die "$ARGV[0]: $!";
+$csv->header($fh);
+
+# Load parishes from supplied CSV file.
+while (my $row = $csv->getline_hr($fh)) {
+    my $name = $row->{name};
+    my $body = FixMyStreet::DB->resultset('Body')->find({ name => $name });
+    if (!$body) {
+        die "Error: Couldn't find body called $name\n";
+    }
+
+    push @parishes, { name => $name, email => $row->{email}, body => $body };
+}
+
+# Create categories for parishes.
+foreach my $parish (@parishes) {
+    foreach my $category_name (@$categories) {
+        my $contact_rs = FixMyStreet::DB->resultset('Contact')->search({
+            body_id => $parish->{body}->id,
+            category => $category_name,
+        });
+
+        if ($contact_rs->count > 0) {
+            print "Existing contact found...skipping\n";
+            next;
+        }
+
+        print "Creating category $category_name for $parish->{name}...";
+
+        my $contact = $contact_rs->create({
+            email => $parish->{email},
+            state => 'confirmed',
+            editor => basename($0),
+            whenedited => \'current_timestamp',
+            note => "created automatically by script",
+        });
```

Comment:
It would be good to have a function on Contact that did this update, so you'd know to supply editor/note and it could add whenedited automatically, used in a few places I reckon.

Reply:
Have added a method to the Contact ResultSet in e6304e4d04c17e0487ae28defd901dcd1ee46949 to help with adding a note when creating a contact.